CharlieBackend.Api.UnitTes/ScheduleServiceTests.cs

```diff
@@ -0,0 +1,649 @@
+        private CreateScheduleDto validScheduleDTO;
+        private EventOccurrence validEventOccurrence;
+
+        public ScheduleServiceTests()
+        {
+            _mapper = GetMapper(new ModelMappingProfile());
+            _scheduledEventFactory = new ScheduledEventHandlerFactory();
+            _scheduleRepositoryMock = new Mock<IScheduledEventRepository>();
+            _eventOccuranceRepositoryMock = new Mock<IEventOccurrenceRepository>();
+            _unitOfWorkMock.Setup(x => x.ScheduledEventRepository).Returns(_scheduleRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.EventOccurrenceRepository).Returns(_eventOccuranceRepositoryMock.Object);
+            validScheduleDTO = new CreateScheduleDto
+            {
+                Pattern = new PatternForCreateScheduleDTO
+                {
+                    Type = PatternType.Daily,
+                    Interval = 1,
+                    DaysOfWeek = new List<DayOfWeek> { DayOfWeek.Monday },
+                },
+
+                Range = new OccurenceRange
+                {
+                    StartDate = DateTime.Parse("Jan 1, 2021"),
+                    FinishDate = DateTime.Parse("Feb 1, 2021")
+                },
+
+                Context = new ContextForCreateScheduleDTO
+                {
+                    GroupID = 3,
+                    MentorID = 1,
+                    ThemeID = 5
+                }
+            };
+
+            validEventOccurrence = new EventOccurrence
+            {
+                Pattern = validScheduleDTO.Pattern.Type,
+                StudentGroupId = validScheduleDTO.Context.GroupID,
+                EventStart = validScheduleDTO.Range.StartDate,
+                EventFinish = validScheduleDTO.Range.FinishDate.Value,
+                Storage = EventOccuranceStorageParser.GetPatternStorageValue(validScheduleDTO.Pattern)
+            };
+        }
+
+        private void Initialize(CreateScheduleDto createScheduleDto, Mock<IThemeRepository> themeRepositoryMock = null,
+            Mock<IMentorRepository> mentorRepositoryMock = null, Mock<IStudentGroupRepository> studentGroupRepositoryMock = null)
+        {
+            _scheduleRepositoryMock.Setup(x => x.AddRange(new List<ScheduledEvent>()
+            {
+                new ScheduledEvent
+                {
+                    ThemeId = 5,
+                    StudentGroupId = 3,
+                    MentorId = 1
+                }
+            }
+            ));
+
+            _eventOccuranceRepositoryMock.Setup(x => x.Add(
+                new EventOccurrence
+                {
+                    StudentGroupId = 3,
+                    EventStart = createScheduleDto.Range.StartDate,
+                    Pattern = createScheduleDto.Pattern.Type
+                }
+                ));
+
+            if (themeRepositoryMock != null)
+            {
+                _unitOfWorkMock.Setup(x => x.ThemeRepository).Returns(themeRepositoryMock.Object);
```

Comment:
Move this to the constructor. It will simplify this method.

Reply:
Done